src/server/acl/acl_family.cc

```diff
@@ -157,12 +163,31 @@ std::variant<User::UpdateRequest, ErrorReply> ParseAclSetUser(CmdArgList args) {
 
 }  // namespace
 
+void AclFamily::StreamUpdatesToAllProactorConnections(std::string_view user, uint32_t update_cat) {
+  auto update_cb = [user, update_cat]([[maybe_unused]] size_t id, util::Connection* conn) {
+    DCHECK(conn);
+    auto connection = static_cast<facade::Connection*>(conn);
+    auto ctx = static_cast<ConnectionContext*>(connection->cntx());
+    if (ctx && user == ctx->authed_username) {
+      ctx->acl_categories = update_cat;
+    }
+  };
+
+  pp_.AwaitFiberOnAll([this, update_cb](util::ProactorBase* pb) {
+    for (auto& listener : listeners_) {
+      listener->TraverseConnections(update_cb);
+    }
+  });
+}
+
 void AclFamily::SetUser(CmdArgList args, ConnectionContext* cntx) {
   std::string_view username = facade::ToSV(args[0]);
   auto req = ParseAclSetUser(args.subspan(1));
   auto error_case = [cntx](ErrorReply&& error) { (*cntx)->SendError(error); };
-  auto update_case = [username, cntx](User::UpdateRequest&& req) {
+  auto update_case = [username, cntx, this](User::UpdateRequest&& req) {
     ServerState::tlocal()->user_registry->MaybeAddAndUpdate(username, std::move(req));
+    auto cred = ServerState::tlocal()->user_registry->GetCredentials(username);
+    StreamUpdatesToAllProactorConnections(username, cred.acl_categories);
```

Comment:
1. maybe you want to do this only if this is not a new user?
2. do we want to have a lock here so no 2 set user commands are done at the same time because if we have 2 set commands for the same user at the same time what will be set in the connections?

Reply:
1. maybe you want to do this only if this is not a new user? 
That's true we can do this
2. That's a good catch! I already have a pattern for this :stuck_out_tongue_winking_eye: